components/bl-worldClock-component/src/views/digital-types/digital-clock-flip.js

```diff
@@ -0,0 +1,126 @@
+import { useEffect } from 'react';
+
+import { useObjectState } from '../helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+const Formats = {
+  AM  : 'AM',
+  PM  : 'PM',
+  AMPM: 'ampm',
+  HOUR: 'hour',
+};
+const LimitValues = {
+  MAX_HOUR       : 23,
+  MAX_TIME_UNIT  : 59,
+  MIN_DIGIT_VALUE: 10,
+};
+const FOLD = 'fold';
+const UNFOLD = 'unfold';
+
+export function DigitalClockFlip({ time, displaySeconds }) {
+  const { hour, minute, second, isAmpm, ampm } = time;
+
+  const [state, setState] = useObjectState({
+    hour, minute, second, ampm,
+    hourShuffle  : true,
+    minuteShuffle: true,
+    secondShuffle: true,
+    ampmShuffle  : true,
+  });
+
+  useEffect(() => {
+    if (hour !== state.hour) {
+      setState({ hour, hourShuffle: !state.hourShuffle });
+    }
+
+    if (minute !== state.minute) {
+      setState({ minute, minuteShuffle: !state.minuteShuffle });
+    }
+
+    if (second !== state.second) {
+      setState({ second, secondShuffle: !state.secondShuffle });
+    }
+
+    if (ampm !== state.ampm) {
+      setState({ ampm, ampmShuffle: !state.ampmShuffle });
+    }
+  }, [hour, minute, second, ampm]);
+
+  return (
+    <div className="flip-clock">
+      <FlipUnitContainer unit="hour" digit={ state.hour } shuffle={ state.hourShuffle }/>
+      <Semicolon/>
+      <FlipUnitContainer unit="minute" digit={ state.minute } shuffle={ state.minuteShuffle }/>
+
+      { displaySeconds && (
+        <>
+          <Semicolon/>
+          <FlipUnitContainer unit="second" digit={ state.second } shuffle={ state.secondShuffle }/>
+        </>
+      ) }
+
+      { isAmpm && (
+        <FlipUnitContainer unit="ampm" digit={ state.ampm } shuffle={ state.ampmShuffle }/>
+      ) }
+    </div>
+  );
+}
+
+function Card({ className, digit }) {
+  return (
+    <div className={ className }>
+      <span>{ digit }</span>
+    </div>
+  );
+}
+
+function FlipUnitContainer({ digit, shuffle, unit }) {
+  let currentDigit, previousDigit;
+
+  if (unit === Formats.AMPM) {
+    currentDigit = digit;
+    previousDigit = digit === Formats.AM ? Formats.PM : Formats.AM;
+  } else {
+    currentDigit = Number(digit);
+    previousDigit = digit - 1;
+
+    [previousDigit, currentDigit] = ensureLimits(previousDigit, currentDigit, unit);
```

Comment:
I think it's too should work, as we have mutation inside the ensureLimits function

```suggestion
    ensureLimits(previousDigit, currentDigit, unit);
```

Reply:
Sorry, that doesn't work.